### PR TITLE
feat: parallelize tests

### LINKED ISSUE #139: Parallelize tests
**Describe the solution you'd like**
Run the tests in parallel, for improved isolation (create a new cluster for each test) and performance.

**Anything else you would like to add:** 

**Environment:**

- Eraser version: v0.1.0
- Kubernetes version: (use `kubectl version`): N/A

### PR DESCRIPTION
**Pull request by pmengelbert**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #139

**Special notes for your reviewer**:

### CODE REVIEW

File: test/e2e/imagelist_change/eraser_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestRemoveImagesFromAllNodes(t *testing.T) {
+	const (
+		nginx         = "nginx"
+		nginxLatest   = "docker.io/library/nginx:latest"
```
Comment (sozercan) [inline test/e2e/imagelist_change/eraser_test.go]: seems like we are duplicating these, should we keep a common file?

File: test/e2e/imagelist_skip_nodes/main_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		envfuncs.LoadDockerImageToCluster(util.KindClusterName, managerImage),
+		envfuncs.LoadDockerImageToCluster(util.KindClusterName, image),
+		deployEraserManifest(eraserNamespace),
+	).Finish(
+		envfuncs.DeleteNamespace(namespace),
+	)
+	os.Exit(testenv.Run(m))
+}
+
+func deployEraserManifest(namespace string) env.Func {
```
Comment (sozercan) [inline test/e2e/imagelist_skip_nodes/main_test.go]: same here for duplicate

File: test/e2e/collector_pipeline/eraser_test.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestRemoveImagesFromAllNodes(t *testing.T) {
```
Comment (sozercan) [inline test/e2e/collector_pipeline/eraser_test.go]: do we still need these to be exported?

File: test/e2e/imagelist_prune_images/main_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/envfuncs"
+)
+
+func TestMain(m *testing.M) {
+	utilruntime.Must(eraserv1alpha1.AddToScheme(scheme.Scheme))
+
+	util.Testenv = env.NewWithConfig(envconf.New())
+	// Create KinD Cluster
+	namespace := envconf.RandomName("eraser-ns", 16)
```
Comment (sozercan) [inline test/e2e/imagelist_prune_images/main_test.go]: looks like we can also make `eraser-ns` a common const? can we reuse `EraserNamespace`  here?

File: test/e2e/imagelist_change/eraser_test.go (hunk 1 above)
Comment (sozercan) [reply test/e2e/imagelist_change/eraser_test.go:27]: can we reuse `EraserNamespace`  here?

File: test/e2e/collector_disable_scan/eraser_test.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+	disableScanFeat := features.New("Test Scanner Disabled Prune").
+		Assess("ImageCollector CR is generated", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			c, err := cfg.NewClient()
+			if err != nil {
+				t.Error("Failed to create new client", err)
+			}
+
+			imagecollector := eraserv1alpha1.ImageCollector{}
+			wait.For(func() (bool, error) {
+				err := c.Resources().Get(ctx, "imagecollector-shared", "default", &imagecollector)
```
Comment (sozercan) [inline test/e2e/collector_disable_scan/eraser_test.go]: same here for `imagecollector-shared`

File: test/e2e/imagelist_alias/eraser_test.go [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+				t.Error("failed to tag nginx image", err)
+			}
+
+			// Load the images into the cluster
+			_, err = util.KindLoadImage(util.KindClusterName, nginxAliasOne)
+			if err != nil {
+				t.Error("failed to load kind image", err)
+			}
+
+			_, err = util.KindLoadImage(util.KindClusterName, nginxAliasTwo)
```
Comment (sozercan) [inline test/e2e/imagelist_alias/eraser_test.go]: can we pass multiple images to `KindLoadImage`? it's not in the PR scope, we can create an issue for this too

File: test/e2e/collector_disable_scan/main_test.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+	util.Testenv = env.NewWithConfig(envconf.New())
+	// Create KinD Cluster
+	namespace := envconf.RandomName("eraser-ns", 16)
+	util.Testenv.Setup(
+		envfuncs.CreateKindClusterWithConfig(util.KindClusterName, util.NodeVersion, "../kind-config.yaml"),
+		envfuncs.CreateNamespace(namespace),
+		envfuncs.LoadDockerImageToCluster(util.KindClusterName, util.ManagerImage),
+		envfuncs.LoadDockerImageToCluster(util.KindClusterName, util.Image),
+		envfuncs.LoadDockerImageToCluster(util.KindClusterName, util.ScannerImage),
```
Comment (sozercan) [inline test/e2e/collector_disable_scan/main_test.go]: can we save time by not loading scanner image?

File: test/e2e/collector_disable_scan/eraser_test.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+				if imagecollector.ObjectMeta.Name == "imagecollector-shared" {
+					return true, nil
+				}
+
+				return false, nil
+			}, wait.WithTimeout(time.Minute*3))
+
+			return ctx
+		}).
+		Assess("ImageList Spec Contains Same Images As ImageCollecotor Shared", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
```
Comment (sozercan) [inline test/e2e/collector_disable_scan/eraser_test.go]:
```suggestion
		Assess("ImageList Spec Contains Same Images As ImageCollector Shared", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
```

typo

File: test/e2e/collector_pipeline/eraser_test.go (hunk 3 above)
Comment (pmengelbert) [reply test/e2e/collector_pipeline/eraser_test.go]: I believe testing functions have to be public, but I'll check.

File: test/e2e/collector_pipeline/eraser_test.go [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+	const (
+		alpine = "alpine"
+	)
+
+	collectScanErasePipelineFeat := features.New("Test Remove Image From All Nodes").
```
Comment (sozercan) [inline test/e2e/collector_pipeline/eraser_test.go]: can we describe this test

File: test/e2e/imagelist_rm_images/eraser_test.go [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+	"k8s.io/apimachinery/pkg/labels"
+
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestRemoveImagesFromAllNodes(t *testing.T) {
+	rmImageFeat := features.New("Test Remove Image From All Nodes").
```
Comment (sozercan) [inline test/e2e/imagelist_rm_images/eraser_test.go]: can we describe this test

File: test/e2e/imagelist_prune_images/eraser_test.go [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+	"k8s.io/apimachinery/pkg/labels"
+
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestRemoveImagesFromAllNodes(t *testing.T) {
+	pruneImagesFeat := features.New("Remove all non-running images from cluster").
```
Comment (sozercan) [inline test/e2e/imagelist_prune_images/eraser_test.go]: can we add that this is prune test

**Review by sozercan — APPROVED** [review]
added a few comments regarding test descriptions, otherwise LGTM. Thank you!

File: test/e2e/imagelist_prune_images/eraser_test.go (hunk 11 above)
Comment (sozercan) [reply test/e2e/imagelist_prune_images/eraser_test.go]: perhaps update the names for each test to match?

**Pull request merged**